src/forwarder.rs

```diff
@@ -71,21 +103,57 @@ impl Hash for Key {
 /// It maintains a collection of tokio task handles, each one identified by a [`Key`] containing
 /// the connection information and responsible for providing forwarder functionalities. For
 /// instance, a task could open a remote terminal between the device and a certain host.
-#[derive(Debug, Default)]
+#[derive(Debug)]
 pub struct Forwarder {
+    tx_state: Sender<SessionState>,
     tasks: HashMap<Key, JoinHandle<()>>,
 }
 
+const CHANNEL_STATE_SIZE: usize = 50;
```

Comment:
Move this definition on top, after the `use` but before the structs declaration.